Delete incidents, reports and weaponinfo via Right Click

LeSiiN opened the pull request:
- this should allow ppl to delete incidents, reports and weapon infos ( almost the same as delete bolo )
- Preview: https://www.youtube.com/watch?v=QxXxUi611gE

LeSiiN committed:
added mdt:server:deleteIncidents event

- added mdt:server:deleteIncidents event to server > main.lua

LeSiiN committed:
added delete incidents

added delete incidents

LeSiiN committed:
added .incidents-delete

added .incidents-delete

LeSiiN committed:
deleteIncidents Callback

deleteIncidents Callback

LeSiiN committed:
deleteReports callback

deleteReports callback

LeSiiN committed:
added mdt:server:deleteReports event

added mdt:server:deleteReports event

LeSiiN committed:
added delete report stuff

added delete report stuff

LeSiiN committed:
added .reports-delete:hover

added .reports-delete:hover

LeSiiN committed:
deleteWeapons callback

deleteWeapons callback

LeSiiN committed:
mdt:server:deleteWeapons added

mdt:server:deleteWeapons added

LeSiiN committed:
.weapons-delete added

.weapons-delete added

LeSiiN committed:
added id to weapon info and added Delete Weapon Info with right click

- added id to weapon info
- added Delete Weapon Info

BackSH00TER commented on server/main.lua:
```diff
@@ -567,6 +567,45 @@ RegisterNetEvent('mdt:server:newBolo', function(existing, id, title, plate, owne
+		local Player = QBCore.Functions.GetPlayer(src)
+		local JobType = GetJobType(Player.PlayerData.job.name)
+		if JobType == 'police' then
+			local fullname = Player.PlayerData.charinfo.firstname .. ' ' .. Player.PlayerData.charinfo.lastname
```
nit: `fullName` camelcase

LeSiiN replied:
done <3

BackSH00TER commented on server/main.lua:
```diff
@@ -567,6 +567,45 @@ RegisterNetEvent('mdt:server:newBolo', function(existing, id, title, plate, owne
+		local JobType = GetJobType(Player.PlayerData.job.name)
+		if JobType == 'police' then
+			local fullname = Player.PlayerData.charinfo.firstname .. ' ' .. Player.PlayerData.charinfo.lastname
+			MySQL.update("DELETE FROM `mdt_reports` WHERE id=:id", { id = id, jobtype = JobType })
```
Shouldn't need to pass jobType as a param here since your looking just for a match on id
```suggestion
			MySQL.update("DELETE FROM `mdt_reports` WHERE id=:id", { id = id })

```

LeSiiN replied:
done <3

BackSH00TER commented on server/main.lua:
```diff
@@ -567,6 +567,45 @@ RegisterNetEvent('mdt:server:newBolo', function(existing, id, title, plate, owne
+		local JobType = GetJobType(Player.PlayerData.job.name)
+		if JobType == 'police' then
+			local fullname = Player.PlayerData.charinfo.firstname .. ' ' .. Player.PlayerData.charinfo.lastname
+			MySQL.update("DELETE FROM `mdt_incidents` WHERE id=:id", { id = id, jobtype = JobType })
```
same comments here

LeSiiN replied:
done <3

BackSH00TER commented on ui/app.js:
```diff
@@ -4510,6 +4510,31 @@ $(document).ready(() => {
       });
+      $(".contextmenu").on("click", ".incidents-delete", function () {
+        $(".incidents-items")
+          .find("[data-id='" + $(this).data("info") + "']")
```
nit: using back ticks can make this a bit simpler
```suggestion
          .find(`[data-id="${$(this).data("info")}"]`)

```

LeSiiN replied:
done <3

BackSH00TER commented on ui/app.js:
```diff
@@ -4851,6 +4876,31 @@ $(document).ready(() => {
       });
+      $(".contextmenu").on("click", ".reports-delete", function () {
+        $(".reports-items")
+          .find("[data-id='" + $(this).data("info") + "']")
```
same nit here

LeSiiN replied:
done <3

BackSH00TER commented on ui/app.js:
```diff
@@ -4964,6 +5014,31 @@ $(document).ready(() => {
       $(".weapon-info-imageurl-input").val(table["image"]);
+      $(".contextmenu").on("click", ".weapons-delete", function () {
+        $(".weapons-items")
+          .find("[data-id='" + $(this).data("info") + "']")
```
same nit

LeSiiN replied:
done <3

BackSH00TER reviewed (commented):
Couple of nits

LeSiiN committed:
changed suggested changes

changed suggested changes

LeSiiN committed:
added suggested changes

added suggested changes

LeSiiN committed:
added rank check

- added rank check for removal of incidents,reports or weapon data
- little issue i got, it sends it 3 times into the logs
